### DISCUSSION TITLE
How to enable availability for select devices only?

### DISCUSSION THREAD

**Discussion by ovizii** [Koenkk/zigbee2mqtt#12786] [General]
I did find these instructions but can't get it to work: 

https://www.zigbee2mqtt.io/guide/configuration/device-availability.html#availability-advanced-configuration 

inside configuration.yaml I find 

```
devices:
  - devices.yaml
```

and 

`availability: false`

so this looks OK as I only want select devices to have availability enabled. I then go and edit devices.yaml and add availability to one device:

```
'0x60a423fffead7ba2':
  friendly_name: CellarPlugDry
  availability: true
'0x00158d00080d08cc':
  friendly_name: CommanderB
  legacy: false
```

I saved the file, restarted HA, restarted the zigbee2mqtt add-on and nothing, that device still shows availability: Disabled in zigbee2mqtt 

I know the file works because the legacy: false setting get picked up by zigbee2mqtt, so I must be doing something wrong?

![image](https://user-images.githubusercontent.com/837902/173186552-94daac10-980b-4988-ae5b-513efcd247a3.png)

changes in devices.yaml reflect in the gui:
![image](https://user-images.githubusercontent.com/837902/173186631-c5201302-1698-49ae-97e5-9757ee05a25c.png)

This never changes according to what I add to devices.yaml:
![image](https://user-images.githubusercontent.com/837902/173186720-a6a35817-4bad-44e0-90b9-33d9f6bb2660.png)

**Comment by Koenkk** [comment]
Your config appears correct but I think it doesn't show up correctly in the GUI. Test if the availability feature is enabeld by removing one of the plugs from mains power, after some time a message should be published to e.g. `zigbee2mqtt/CellarPlugDry/availability` `offline`.

**Comment by ovizii** [comment]
Thanks for your tip, I will give it a try, but I admit I have no idea how to check for that message being published. I guess I'll see if I can figure it out. 
For now, I tried creating an automation to test, but it does not seem to be firing. I will keep trying. Feel free to close this thread or comment if you have more tips for me.

![image](https://user-images.githubusercontent.com/837902/173235506-fc828dda-9839-48c6-99d1-b0472c1f1129.png)

**Comment by ovizii** [comment]
I take everything back, the automation fired with slight delay. All is well. 

You were right, zigbee2mqtt's GUI simply did not display "availability: true" in its GUI.

**Reply by MikeVRT** [reply]
Yes, it's error in GUI. In mqtt topics devices with "availability: true" option appears.

**Comment by bennybubble** [comment]
Hi,

I tried to add 'availability: true' to my configuration.yaml file but after 'restart zigbee2mqtt' the setting disappeared.

  '0x0015bc0034003d85':
      friendly_name: brandmelder1
      description: woonkamer
      availability: true

`Zigbee2MQTT version [2.2.1]   commit: [3a474187]
Coordinator type                      zStack12
Coordinator revision                 20211116
Coordinator IEEE Address         0x00124b001cd482c2
Frontend version                      0.9.7
zigbee-herdsman-converters version    23.20.1
zigbee-herdsman version          3.4.11

`

Any hint?
-ben
